Lambda field (functional interface) deserialize failed with CompatiableFieldSerializer

**Describe the bug**
We have a class with a field to store lambda, the field's class is a functional interface, something like this:
```java
public class TestClass implements Serializable {
    private static final long serialVersionUID = 4061358520297707007L;

    @FunctionalInterface
    public interface Callback extends Serializable {
        void execute();
    }

    private Callback callback;
}
```
When serialize then deserialize this class with CompatiableFieldSerializer, we encountered this exception:
```
Caused by: com.esotericsoftware.kryo.KryoException: Read type is incompatible with the field type: com.esotericsoftware.kryo.serializers.ClosureSerializer$Closure -> com.esotericsoftware.kryo.serializers.ClosureSerializer$Closure (com.xxx.TestClass#callback)
```
I believe it's because CompatiableFieldSerializer cached the field's value class (set to TestClass#callback) during serialization, but the actual output content is ClosureSerializer$Closure. Then during deserialization, it's mismatched.
**To Reproduce**
Use CompatiableFieldSerializer to serialize a class with lambda field(s), then deserialize it.

**Environment:**
 - OS: Ubuntu 20.04
 - JDK Version: 8
 - Kryo Version: 5.0.4

**Additional context**
I switch to FieldSerializer it works fine, but I didn't test TaggedFieldSerializer and VersionFieldSerializer.

--- a/src/com/esotericsoftware/kryo/util/Util.java
+++ b/src/com/esotericsoftware/kryo/util/Util.java
@@ -23,6 +23,7 @@ import static com.esotericsoftware.minlog.Log.*;
 
 import com.esotericsoftware.kryo.Serializer;
 import com.esotericsoftware.kryo.SerializerFactory;
+import com.esotericsoftware.kryo.serializers.ClosureSerializer;
 import com.esotericsoftware.kryo.serializers.FieldSerializer;
 import com.esotericsoftware.kryo.util.Generics.GenericType;
 
@@ -232,8 +233,9 @@ public class Util {
 	public static boolean isAssignableTo (Class<?> from, Class<?> to) {
 		if (to == Object.class) return true;
 		if (to.isAssignableFrom(from)) return true;
-		if (from.isPrimitive()) return isPrimitiveWrapperOf(to, from);
 		if (to.isPrimitive()) return isPrimitiveWrapperOf(from, to);
+		if (from.isPrimitive()) return isPrimitiveWrapperOf(to, from);
+		if (from == ClosureSerializer.Closure.class) return to.isInterface();
 		return false;
 	}
 

--- a/src/com/esotericsoftware/kryo/util/Util.java
+++ b/src/com/esotericsoftware/kryo/util/Util.java
@@ -233,8 +233,8 @@ public class Util {
 	public static boolean isAssignableTo (Class<?> from, Class<?> to) {
 		if (to == Object.class) return true;
 		if (to.isAssignableFrom(from)) return true;
-		if (to.isPrimitive()) return isPrimitiveWrapperOf(from, to);
 		if (from.isPrimitive()) return isPrimitiveWrapperOf(to, from);
+		if (to.isPrimitive()) return isPrimitiveWrapperOf(from, to);
 		if (from == ClosureSerializer.Closure.class) return to.isInterface();
 		return false;
 	}

--- a/test/com/esotericsoftware/kryo/serializers/CompatibleFieldSerializerTest.java
+++ b/test/com/esotericsoftware/kryo/serializers/CompatibleFieldSerializerTest.java
@@ -29,6 +29,7 @@ import com.esotericsoftware.kryo.io.Input;
 import com.esotericsoftware.kryo.io.Output;
 
 import java.io.Serializable;
+import java.lang.invoke.SerializedLambda;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
@@ -476,6 +477,19 @@ class CompatibleFieldSerializerTest extends KryoTestCase {
 		assertEquals(o2, kryo.readClassAndObject(new Input(output2.getBuffer())));
 	}
 
+	@Test
+	void testClassWithLambdaField () {
+		CompatibleFieldSerializer.CompatibleFieldSerializerConfig config = new CompatibleFieldSerializer.CompatibleFieldSerializerConfig();
+		config.setChunkedEncoding(true);
+		config.setReadUnknownFieldData(true);
+		kryo.setDefaultSerializer(new CompatibleFieldSerializerFactory());
+		kryo.register(ClassWithLambdaField.class);
+		kryo.register(SerializedLambda.class);
+		kryo.register(ClosureSerializer.Closure.class, new ClosureSerializer());
+
+		roundTrip(236, new ClassWithLambdaField());
+	}
+
 	public static class TestClass {
 		public String text = "something";
 		public int moo = 120;
@@ -730,6 +744,27 @@ class CompatibleFieldSerializerTest extends KryoTestCase {
 			if (o == null || getClass() != o.getClass()) return false;
 			final ClassWithLongField that = (ClassWithLongField)o;
 			return Objects.equals(value, that.value);
+		}
+	}
+
+	public static class ClassWithLambdaField {
+
+		@FunctionalInterface
+		public interface Callback extends Serializable {
+			int call();
+		}
+
+		private final Callback callback;
+
+		public ClassWithLambdaField () {
+			callback = () -> 1;
+		}
+
+		public boolean equals (Object o) {
+			if (this == o) return true;
+			if (o == null || getClass() != o.getClass()) return false;
+			ClassWithLambdaField that = (ClassWithLambdaField)o;
+			return Objects.equals(callback.call(), that.callback.call());
 		}
 	}
 }

